pennylane/data/dataset.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`~pennylane.data.Dataset` class and its associated functions.
+"""
+
+from abc import ABC
+import dill
+import zstd
+
+
+class Dataset(ABC):
+    """Create a dataset object.
+    A dataset might be a collection of information that describes a physical system and its evolution.
+    For example, a dataset for an arbitrary quantum system could have a Hamiltonian, its ground state,
+    and an efficient state-preparation circuit for that state.
+
+    Args:
+        dtype (string): the type of dataset, e.g., `qchem`, `qspin`, etc.
```

Comment:
Is having  "e.g., ..." consistent with the rest of PL?

Also consider: the type of the dataset

Reply:
"e.g." is not common for Args. Edit Made.

I didn't understand "consider the type of the dataset". Is this about whether it is a `qchem` or `qspin` dataset? That should be covered by the `dtype`.

As for what we get when we do `print(type(dataset))`. That's <class 'pennylane.data.dataset.Dataset'>

Reply:
No, just the phrasing: "the type of _**the**_ dataset"

Reply:
Oh! Of course :laughing: 
Edit made :+1: